Feature/se 46

Arkcorner opened the pull request:
closes SE2-WS23/documentation#46 

![git meme](https://github.com/SE2-WS23/Skillverse-Web/assets/115979577/d17abda4-9367-482b-b59d-240a505dd32b)

Arkcorner committed:
Employee Creation Page

Arkcorner committed:
fix Sonarcloud

Arkcorner committed:
Merge branch 'main' into feature/se-46

Arkcorner committed:
enhancement Employee creation form

Arkcorner committed:
Merge branch 'feature/se-46' of https://github.com/SE2-WS23/Skillverse-Web into feature/se-46

Arkcorner committed:
fix use state

Arkcorner committed:
Merge branch 'main' into feature/se-46

Arkcorner committed:
Update src/pages/EmployeeCreation/EmployeeCreationPage.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
Update src/pages/EmployeeCreation/EmployeeCreationPage.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
Update src/pages/EmployeeCreation/EmployeeCreationPage.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
Update src/pages/EmployeeCreation/components/UploadBox.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
Update src/pages/EmployeeCreation/components/UploadedFile.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
fix Formatting/ div + input now MUI

Arkcorner committed:
Update src/pages/EmployeeCreation/EmployeeCreationPage.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
Update src/pages/EmployeeCreation/EmployeeCreationPage.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
Update src/pages/EmployeeCreation/EmployeeCreationPage.jsx

Co-authored-by: Julius Becker <julix.becker@gmail.com>

Arkcorner committed:
update

Arkcorner committed:
Merge branch 'feature/se-46' of https://github.com/SE2-WS23/Skillverse-Web into feature/se-46

Arkcorner committed:
Merge branch 'main' into feature/se-46

Chrissydrx commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
Layout is broken for small devices. Avatar is stretched.
![image](https://github.com/SE2-WS23/Skillverse-Web/assets/37250473/36bed9c0-b996-4a20-9727-4f2b0fbbff03)

Arkcorner replied:
The Layout for small devices has been fixed

Chrissydrx replied:
![image](https://github.com/SE2-WS23/Skillverse-Web/assets/37250473/4a2b9222-1d03-43a1-ae2c-0526caff5630)
Still not completely fixed for width 900px

Arkcorner replied:
fixed

Arkcorner replied:
<img width="428" alt="Screenshot 2023-12-13 at 17 24 38" src="https://github.com/SE2-WS23/Skillverse-Web/assets/115979577/e1b19147-99f0-402a-ada0-86c2e610eb1a">

Chrissydrx commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
You don't use our prettier settings

Arkcorner replied:
I now use the Prettier settings

Chrissydrx commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
```diff
@@ -0,0 +1,157 @@
+  };
+ const handleChangeAvatar = (event) => {
+    console.log("Avatar has been changed");
+  }
```
Please add space between those functions

Arkcorner replied:
theres now spaces between the functions

Chrissydrx commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
```diff
@@ -0,0 +1,157 @@
+        >
+          <Box maxWidth="50%">
+            <Box>
+              <Avatar sx={{ width: "10vw", height: "20vh", variant: "circular" }} />
```
Weird width and height settings for avatar

Arkcorner replied:
the sizes are now both dependent on the width of the screen and i didnt really see a convention from us for this so i tried to make it just look good

Chrissydrx commented on src/pages/EmployeeCreation/components/UploadBox.jsx:
```diff
@@ -0,0 +1,54 @@
+  */
+
+function UploadBox() {
+  const [selectedFiles, setSelectedFiles] = useState([]);
```
That state can not be accessed in your page component

Chrissydrx replied:
Still not fixed

Arkcorner replied:
fixed

Chrissydrx commented on src/pages/EmployeeCreation/components/UploadBox.jsx:
```diff
@@ -0,0 +1,54 @@
+
+function UploadBox() {
+  const [selectedFiles, setSelectedFiles] = useState([]);
+
+  const handleFileSubmit = (event) => {
+    const files = event.target.files;
+    const newFiles = Array.from(files);
+
+    setSelectedFiles((prevFiles) => [...prevFiles, ...newFiles]);
+    console.log({ selectedFiles });
+  };
```
This seems like you can upload multiple files at once, but you can only chose one file in the picker

Arkcorner replied:
Yes files need to be uploaded one at a time array is just to allow multiple uploads one after the other

Arkcorner replied:
Fixed the Input component of material UI mad this way harder as it didn´t have the multiple property defined so i switched to a normal input field to more suit the needs of our website as its hidden anyway

Arkcorner committed:
Merge branch 'main' into feature/se-46

Chrissydrx commented on src/pages/EmployeeCreation/components/UploadBox.jsx:
```diff
@@ -0,0 +1,54 @@
+  return (
+    <Box>
+      {selectedFiles.map((file) => (
+        <Box key={file.name} display="flex" alignItems="center" mb={1}>
```
Use as key an uuid

Arkcorner replied:
did

Chrissydrx commented on src/pages/EmployeeCreation/components/UploadedFile.jsx:
```diff
@@ -0,0 +1,36 @@
+  return (
+    <Box display="flex" alignItems="center" flexDirection="row" mb={1}>
+      <InsertDriveFile fontSize="small" style={{ marginRight: 8 }} />
+      <Typography variant="subtitle1">Selected File: {fileName}</Typography>
```
Set default for filename

Arkcorner replied:
did

Chrissydrx commented on src/pages/EmployeeCreation/components/UploadedFile.jsx:
```diff
@@ -0,0 +1,36 @@
+*/
+function UploadedFile({ fileName, onDelete }) {
+  const deleteFile = () => {
+    onDelete(fileName);
```
Set default

Chrissydrx replied:
or at least a fallback

Arkcorner replied:
default should be set

Chrissydrx commented on src/pages/EmployeeCreation/components/UploadBox.jsx:
```diff
@@ -0,0 +1,54 @@
+
+  const handleFileUpload = () => {
+    document.getElementById("upload-file").click();
+  };
```
Does this click the upload-file input field?

Arkcorner replied:
yes a straight up on click doesnt seem to work and i got this from coPilot

Arkcorner committed:
Fix

Formatter settings have been adjusted, uuidv4 has been implemented, avatar size is now dynamic

Arkcorner committed:
Merge branch 'main' into feature/se-46

Arkcorner committed:
fix spacing between functions

Arkcorner committed:
Merge branch 'feature/se-46' of https://github.com/SE2-WS23/Skillverse-Web into feature/se-46

Arkcorner committed:
set defaults,  fixed layout for 900px

Arkcorner committed:
fix sonarCloud check

Arkcorner committed:
Merge branch 'main' into feature/se-46

Arkcorner committed:
Merge branch 'main' into feature/se-46

Arkcorner committed:
fix useState accessability

Arkcorner committed:
fix sonarCloud

Arkcorner committed:
fix SonarCloud ... again

Arkcorner committed:
Merge branch 'main' into feature/se-46

Arkcorner committed:
Merge branch 'main' into feature/se-46

j0world commented:
If you open Employee overview on a smaller screen you can see that it doesn't open at the top. Others don't do that either but if you want to fix it you can ask Uzoma on how to do that. 
Only being able to upload a single file at a time is already mentioned by Christian.
I can create a new employee without any information which isn't as bad because there's no logic behind the handleSubmit() anyway but if you're ambitious, that shouldn't be too difficult to change.

j0world commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
```diff
@@ -0,0 +1,179 @@
+                sx={{ width: '10vw', height: '10vw', variant: 'circular' }}
+              />
+              <Fab
+                position='absolute'
```
The profile picture scales but the plus button doesn't.

j0world replied:
width changes now, height doesn't seem to scale correctly

Arkcorner replied:
the + in the fab hinders it from scaling correctly for all sizes but in my testing it was ok for the 900 by 900 px and full screen size I'm not sure how to make it completely dynamic

j0world reviewed (changes requested):
If I add too much stuff, either in Files or Extra info, the Create Employee button lingers at the bottom and can't be moved away.
The same for the profile picture but with the top.

j0world commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
```diff
@@ -0,0 +1,179 @@
+            </Box>
+            <Box>
+              <TextField
+                label='Phone number'
```
Please change to "Phone Number" according to title case

j0world commented on src/pages/EmployeeCreation/EmployeeCreationPage.jsx:
```diff
@@ -0,0 +1,179 @@
+              <TextField
+                margin='normal'
+                type='submit'
+                label='Extra info'
```
Please change to "Extra Info" according title case

Arkcorner committed:
Update EmployeeCreationPage

Fixed Title case and Padding Top and Bottom

j0world commented on src/pages/EmployeeCreation/components/UploadBox.jsx:
I clicked random things out of curiosity, please make it possible to upload multiple files and I shall also willingly approve

Arkcorner replied:
did

j0world replied:
indeed works now. 
If I upload a document with too long of a title however, it messes with your padding
<img width="514" alt="Screenshot 2024-01-23 at 18 47 44" src="https://github.com/SE2-WS23/Skillverse-Web/assets/136974808/c6f47e24-6fdd-4474-92c9-87c92e4eb738">

Arkcorner committed:
Update EmployeeCreationPage

Fixed padding between text fields

Arkcorner committed:
fixed padding and made it possible to submit multiple files at once

Arkcorner committed:
fixed issues with long names and padding

Chrissydrx reviewed (approved):
I trust that you guys did a good job. I didn't check the PR, but because I don't have time in the next days, I will approve this, so that I don't block the team
